### PR TITLE
Add generic opamps and TL074xN

### PR DESCRIPTION
**Pull request by pinoaffe**
depends on https://github.com/LibrePCB-Libraries/LibrePCB_Base.lplib/pull/14

### CODE REVIEW

**Comment by ubruhin** [comment]
Thanks!

Regarding the symbol, I would suggest a slightly more compact shape, like this one from Eagle (10.16mm width and height):
grafik

Components:

- Start name with "OpAmp", see [conventions](https://docs.librepcb.org/#libraryconventions-symbols-naming):
  > If reasonable, start with the generic term (e.g. "Supply GND" instead of "GND Supply") to improve navigation in sorted lists (all supply symbols are listed next to each other)
- Component symbol items should have a suffix defined ("A", "B", "C" etc.)
- Is it intended that V- is at top and V+ at bottom of the power symbol? For me that looks a bit strange ;) 
grafik
- The "or COMPONENT" should be removed from the component's default value, it's not very helpful to see "Quad OpAmp" in the schematics...
- In the Quad OpAmp, all symbols are overlapping: 
grafik

**Comment by pinoaffe** [comment]
Thanks for the review! I made most of the changes you requested, but kept the bigger symbol
Is this okay?

File: cmp/0aabc0df-a3fe-4a95-be1f-7be4cccddb2b/component.lp [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ (version "0.1")
+ (created 2018-12-13T00:20:56Z)
+ (deprecated false)
+ (category 6b8cb737-30d9-49e2-831c-34aa909ef85e)
+ (category e05f53ad-1c05-42c1-9124-d0ed3e445a2d)
+ (schematic_only false)
+ (default_value "{{PARTNUMBER or DEVICE}}")
+ (prefix "U")
+ (signal 53f1cc49-614b-494c-829b-49a22a8a09fe (name "V+") (role passive)
+  (required false) (negated false) (clock false) (forced_net "")
```
Comment (ubruhin) [inline cmp/0aabc0df-a3fe-4a95-be1f-7be4cccddb2b/component.lp]:
```suggestion
  (required true) (negated false) (clock false) (forced_net "")
```

File: cmp/0aabc0df-a3fe-4a95-be1f-7be4cccddb2b/component.lp [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+ )
+ (signal bd997edc-b69b-4766-9c55-00b05febeed7 (name "V-") (role passive)
+  (required false) (negated false) (clock false) (forced_net "")
```
Comment (ubruhin) [inline cmp/0aabc0df-a3fe-4a95-be1f-7be4cccddb2b/component.lp]:
```suggestion
  (required true) (negated false) (clock false) (forced_net "")
```

File: cmp/a1253f34-16e6-4bf5-babf-7f76bc615d04/component.lp [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  (required false) (negated false) (clock false) (forced_net "")
+ )
+ (signal 95b6b2d3-dcdd-4336-a0d8-510c1d90bd08 (name "-") (role passive)
+  (required false) (negated false) (clock false) (forced_net "")
+ )
+ (signal d59f30a0-70c8-46f0-bac0-9458d2cf599e (name "+") (role passive)
+  (required false) (negated false) (clock false) (forced_net "")
+ )
+ (signal 89e56ad2-1b11-4fa7-a8c5-f616147af19f (name "V+") (role passive)
+  (required false) (negated false) (clock false) (forced_net "")
```
Comment (ubruhin) [inline cmp/a1253f34-16e6-4bf5-babf-7f76bc615d04/component.lp]:
```suggestion
  (required true) (negated false) (clock false) (forced_net "")
```

File: cmp/a1253f34-16e6-4bf5-babf-7f76bc615d04/component.lp [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+ )
+ (signal d757a373-aeab-43af-a81e-5d7de5fe94d1 (name "V-") (role passive)
+  (required false) (negated false) (clock false) (forced_net "")
```
Comment (ubruhin) [inline cmp/a1253f34-16e6-4bf5-babf-7f76bc615d04/component.lp]:
```suggestion
  (required true) (negated false) (clock false) (forced_net "")
```

File: cmp/f4d5dec7-133c-41c4-b32a-3e8c50efed99/component.lp [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+ (version "0.1")
+ (created 2018-12-13T00:12:31Z)
+ (deprecated false)
+ (category 6b8cb737-30d9-49e2-831c-34aa909ef85e)
+ (category e05f53ad-1c05-42c1-9124-d0ed3e445a2d)
+ (schematic_only false)
+ (default_value "{{PARTNUMBER or DEVICE}}")
+ (prefix "U")
+ (signal 44c28122-827b-4f80-b43f-bee48a8e700f (name "V+") (role passive)
+  (required false) (negated false) (clock false) (forced_net "")
```
Comment (ubruhin) [inline cmp/f4d5dec7-133c-41c4-b32a-3e8c50efed99/component.lp]:
```suggestion
  (required true) (negated false) (clock false) (forced_net "")
```

File: cmp/f4d5dec7-133c-41c4-b32a-3e8c50efed99/component.lp [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+ )
+ (signal 5956a15c-d7d0-4fd2-9248-a502ec69e298 (name "V-") (role passive)
+  (required false) (negated false) (clock false) (forced_net "")
```
Comment (ubruhin) [inline cmp/f4d5dec7-133c-41c4-b32a-3e8c50efed99/component.lp]:
```suggestion
  (required true) (negated false) (clock false) (forced_net "")
```

File: cmp/0aabc0df-a3fe-4a95-be1f-7be4cccddb2b/component.lp [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+ (signal 55051256-bdd6-43da-99b5-80afb795ff66 (name "O4") (role passive)
+  (required false) (negated false) (clock false) (forced_net "")
+ )
+ (variant 52ababe6-3d42-40dc-9108-422ec3637479 (norm "")
+  (name "default")
+  (description "")
+  (gate 19b32e3e-6b03-44b8-b3b9-8bd4a0bb87d9
+   (symbol d759f88a-c0fe-4b76-b5a6-5e85d74ff722)
+   (position 20.32 0.0) (rotation 0.0) (required true) (suffix "A")
+   (pin 0cc9ed65-ea1d-46fa-aafb-2472a0607e23 (signal 8421a7f6-51b9-4e8c-b65c-d07dc316d088) (text pin))
```
Comment (ubruhin) [inline cmp/0aabc0df-a3fe-4a95-be1f-7be4cccddb2b/component.lp]:
```suggestion
   (pin 0cc9ed65-ea1d-46fa-aafb-2472a0607e23 (signal 8421a7f6-51b9-4e8c-b65c-d07dc316d088) (text none))
```

File: cmp/0aabc0df-a3fe-4a95-be1f-7be4cccddb2b/component.lp [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+   (pin 87103c18-5f46-48d2-9553-4f004c7c19e6 (signal 921ca614-e19a-4fa5-8432-2f59dc2c246a) (text pin))
+   (pin f4ec230d-36cf-4b7e-9553-2b65ce39b717 (signal e23711db-4e75-428d-8889-f2bd83364aab) (text pin))
+  )
+  (gate 47affd0b-a4ba-45e6-af67-eab85f1e4d42
+   (symbol d759f88a-c0fe-4b76-b5a6-5e85d74ff722)
+   (position 20.32 20.32) (rotation 0.0) (required true) (suffix "B")
+   (pin 0cc9ed65-ea1d-46fa-aafb-2472a0607e23 (signal a024a8bd-8663-4f58-9787-d658fb22759b) (text pin))
```
Comment (ubruhin) [inline cmp/0aabc0df-a3fe-4a95-be1f-7be4cccddb2b/component.lp]:
```suggestion
   (pin 0cc9ed65-ea1d-46fa-aafb-2472a0607e23 (signal a024a8bd-8663-4f58-9787-d658fb22759b) (text none))
```

File: cmp/0aabc0df-a3fe-4a95-be1f-7be4cccddb2b/component.lp [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+   (pin 87103c18-5f46-48d2-9553-4f004c7c19e6 (signal 458077dc-0b33-4a6b-a600-3d656ef19723) (text pin))
+   (pin f4ec230d-36cf-4b7e-9553-2b65ce39b717 (signal 91945aee-5223-4519-abc5-4dee56ec7ade) (text pin))
+  )
+  (gate db848529-8d84-4814-8d9d-6a5ae1b5b0dc
+   (symbol d759f88a-c0fe-4b76-b5a6-5e85d74ff722)
+   (position 0.0 0.0) (rotation 0.0) (required true) (suffix "C")
+   (pin 0cc9ed65-ea1d-46fa-aafb-2472a0607e23 (signal 4aeb7f5e-0f78-499e-9efc-902aa8b93bb0) (text pin))
```
Comment (ubruhin) [inline cmp/0aabc0df-a3fe-4a95-be1f-7be4cccddb2b/component.lp]:
```suggestion
   (pin 0cc9ed65-ea1d-46fa-aafb-2472a0607e23 (signal 4aeb7f5e-0f78-499e-9efc-902aa8b93bb0) (text none))
```

File: cmp/0aabc0df-a3fe-4a95-be1f-7be4cccddb2b/component.lp [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+   (pin 87103c18-5f46-48d2-9553-4f004c7c19e6 (signal 8c7d2e3d-50f7-4b5e-9e85-dfc869ec333a) (text pin))
+   (pin f4ec230d-36cf-4b7e-9553-2b65ce39b717 (signal 7c1c8635-2e8b-4316-98fc-0ec4ed905fea) (text pin))
+  )
+  (gate 9c64120e-27f3-42e0-9f6b-6467075a58a4
+   (symbol d759f88a-c0fe-4b76-b5a6-5e85d74ff722)
+   (position 0.0 20.32) (rotation 0.0) (required true) (suffix "D")
+   (pin 0cc9ed65-ea1d-46fa-aafb-2472a0607e23 (signal 55051256-bdd6-43da-99b5-80afb795ff66) (text pin))
```
Comment (ubruhin) [inline cmp/0aabc0df-a3fe-4a95-be1f-7be4cccddb2b/component.lp]:
```suggestion
   (pin 0cc9ed65-ea1d-46fa-aafb-2472a0607e23 (signal 55051256-bdd6-43da-99b5-80afb795ff66) (text none))
```

File: cmp/a1253f34-16e6-4bf5-babf-7f76bc615d04/component.lp [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+ )
+ (variant b2f45df6-5f5b-4260-871c-c6378df3c109 (norm "")
+  (name "default")
+  (description "")
+  (gate 77c732a4-5b33-4d3b-acf3-333096dc558e
+   (symbol d759f88a-c0fe-4b76-b5a6-5e85d74ff722)
+   (position 0.0 0.0) (rotation 0.0) (required true) (suffix "A")
+   (pin 0cc9ed65-ea1d-46fa-aafb-2472a0607e23 (signal 50a4a4cc-799b-4413-992f-a8e6822f3826) (text pin))
```
Comment (ubruhin) [inline cmp/a1253f34-16e6-4bf5-babf-7f76bc615d04/component.lp]:
```suggestion
   (pin 0cc9ed65-ea1d-46fa-aafb-2472a0607e23 (signal 50a4a4cc-799b-4413-992f-a8e6822f3826) (text none))
```

File: cmp/f4d5dec7-133c-41c4-b32a-3e8c50efed99/component.lp [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+ (signal 7ebd4fba-aa61-480f-8ff8-a40a4e42861e (name "O2") (role passive)
+  (required false) (negated false) (clock false) (forced_net "")
+ )
+ (variant d212c234-cae0-436e-9c97-6049e2928d33 (norm "")
+  (name "default")
+  (description "")
+  (gate e9f2dab5-0a36-4d8d-aefa-34b2ad3e749f
+   (symbol d759f88a-c0fe-4b76-b5a6-5e85d74ff722)
+   (position 0.0 17.78) (rotation 0.0) (required true) (suffix "A")
+   (pin 0cc9ed65-ea1d-46fa-aafb-2472a0607e23 (signal 44524843-93cb-440e-9baf-8bbc89d9efcb) (text pin))
```
Comment (ubruhin) [inline cmp/f4d5dec7-133c-41c4-b32a-3e8c50efed99/component.lp]:
```suggestion
   (pin 0cc9ed65-ea1d-46fa-aafb-2472a0607e23 (signal 44524843-93cb-440e-9baf-8bbc89d9efcb) (text none))
```

File: cmp/f4d5dec7-133c-41c4-b32a-3e8c50efed99/component.lp [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+   (pin 87103c18-5f46-48d2-9553-4f004c7c19e6 (signal 05150e28-b80e-4ad6-be7a-b4b90dcea1d7) (text pin))
+   (pin f4ec230d-36cf-4b7e-9553-2b65ce39b717 (signal 9d86a1fc-3202-49d6-9a97-bb189270cadc) (text pin))
+  )
+  (gate 3bab07c5-6fbf-47f7-ba7b-94f028f01e87
+   (symbol d759f88a-c0fe-4b76-b5a6-5e85d74ff722)
+   (position 0.0 -17.78) (rotation 0.0) (required true) (suffix "B")
+   (pin 0cc9ed65-ea1d-46fa-aafb-2472a0607e23 (signal 7ebd4fba-aa61-480f-8ff8-a40a4e42861e) (text pin))
```
Comment (ubruhin) [inline cmp/f4d5dec7-133c-41c4-b32a-3e8c50efed99/component.lp]:
```suggestion
   (pin 0cc9ed65-ea1d-46fa-aafb-2472a0607e23 (signal 7ebd4fba-aa61-480f-8ff8-a40a4e42861e) (text none))
```

**Comment by dbrgn** [comment]
> To be honest, I like it much more with the smaller symbol ;) Maybe @dbrgn has an opinion on that?

It's larger than the ADC symbol that we already have in the library:

img

I generally prefer smaller symbols too. The question is whether there's a non-trivial possibility of the pins not named "+" and "-". As long as these names are short, a smaller symbol would work fine. On the other hand, I'm not 100% sure yet if I like the look if we match the shape of the ADC symbol:

img

Here's the alternative if we make it a bit longer (but still the same height as the ADC):

img

Yes, the "O" pin looks crappy because it overlaps with the symbol shape. But we'll have to implement movable labels anyways sooner or later, so we should not make the symbol larger or smaller just because that's still missing.

**Comment by dbrgn** [comment]
By the way, the ADC power pins are called "V+" and "V-", while here they're called "+" and "-". I don't have a strong opinion on which one is better, but I think we should keep things consistent and pick one or the other.

**Comment by pinoaffe** [comment]
I slimmened the symbol and think it can be merged now

**Comment by ubruhin** [comment]
Thanks! Unfortunately one code review suggestion was still not applied, and the symbol was 2.54mm wider than the examples above. I adjusted it by myself, I hope you don't mind ;)

**Pull request merged**